services/ui_backend_service/api/utils.py

```diff
@@ -294,3 +296,28 @@ async def find_records(request: web.BaseRequest, async_table=None, initial_condi
         status, res = format_response_list(
             request, results, page, pagination.pages_total)
         return web_response(status, res)
+
+
+class TTLQueue:
+    def __init__(self, ttl_in_seconds: int):
+        self._ttl: int = ttl_in_seconds
+        self._queue = deque()
+
+    def append(self, value: any):
+        self._queue.append((time.time(), value))
+        self.discard_expired_values()
+
+    def discard_expired_values(self):
+        cutoff_time = time.time() - self._ttl
+        try:
+            while self._queue[0][0] < cutoff_time:
+                self._queue.popleft()
+        except IndexError:
+            pass
```

Comment:
clear choice to keep these synchronous instead of async? Is there any concern that the queue pop might be sluggish if it is packed with events during high utilization times?